pom.xml

```diff
@@ -0,0 +1,166 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (c) 2021 Oracle and/or its affiliates. All rights reserved.
+
+    This program and the accompanying materials are made available under the
+    terms of the Eclipse Public License v. 2.0, which is available at
+    http://www.eclipse.org/legal/epl-2.0.
+
+    This Source Code may also be made available under the following Secondary
+    Licenses when the conditions for such availability set forth in the
+    Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+    version 2 with the GNU Classpath Exception, which is available at
+    https://www.gnu.org/software/classpath/license.html.
+
+    SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+	<parent>
+		<groupId>org.eclipse.ee4j</groupId>
+		<artifactId>project</artifactId>
+		<version>1.0.6</version>
+		<relativePath></relativePath>
+	</parent>
+	<modelVersion>4.0.0</modelVersion>
+	<packaging>jar</packaging>
+
+	<groupId>com.sun.mail</groupId>
+	<artifactId>jakarta.mail-tck</artifactId>
+	<version>1.0.0-SNAPSHOT</version>
+
+	<properties>
+	    <!-- Update next 2 dependencies after releasing Mail-API and Angus -->
+		<mail-api.version>2.1.0-RC1</mail-api.version>
+        <angus.mail.version>1.0.0-M1</angus.mail.version>
+		<jupiter-api.version>5.7.2</jupiter-api.version>
+		<maven.compiler.target>1.8</maven.compiler.target>
+		<maven.compiler.source>1.8</maven.compiler.source>
+		<maven.compiler.plugin>3.8.1</maven.compiler.plugin>
+		<surefire.version>3.0.0-M5</surefire.version>
+		<jar.version>3.2.0</jar.version>
+		<copyright.version>2.3</copyright.version>
+		<sigtest.file>/jakarta.mail_2.0_java_1.8.sig</sigtest.file>
+	</properties>
+
+    <profiles>
+        <profile>
+            <id>jdk9+</id>
+            <activation>
+                <jdk>[9,)</jdk>
+            </activation>
+            <properties>
+	            <sigtest.file>/jakarta.mail_2.0_java_11.sig</sigtest.file>
+	        </properties>
+        </profile>
+    </profiles>
+
+	<dependencies>
+		<dependency>
+			<groupId>org.junit.jupiter</groupId>
+			<artifactId>junit-jupiter</artifactId>
+			<version>${jupiter-api.version}</version>
+			<scope>test</scope>
+		</dependency>
+        <dependency>
+            <groupId>jakarta.mail</groupId>
+            <artifactId>jakarta.mail-api</artifactId>
+            <version>${mail-api.version}</version>
+        </dependency>
+		<dependency>
+			<groupId>org.eclipse.angus</groupId>
+			<artifactId>angus-mail</artifactId>
+			<version>${angus.mail.version}</version>
+		</dependency>
+		<dependency>
+		    <groupId>sigtest</groupId>
+		    <artifactId>sigtest</artifactId>
+		    <version>1.0</version>
+		    <scope>system</scope>
```

Comment:
The Platform TCK is currently using:
```
 <groupId>org.netbeans.tools</groupId>
            <artifactId>sigtest-maven-plugin</artifactId>
            <version>1.4</version>
        </dependency>
```

The XML Binding TCK will also use org.netbeans.tools:sigtest-maven-plugin:1.4 via merged https://github.com/eclipse-ee4j/jaxb-tck/pull/58

Reply:
That version 1.4 does not work with the signature file of JDK11. I get this:
```
[ERROR] sigtest.SignatureTest.test  Time elapsed: 0.057 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: 
/jakarta.mail_2.0_java_11.sig
Can not find or read signature file(s): /tmp/jakarta.mail_2.0_java_11.sig
```

I have debugged it, and the problem is that the signature file is #Signature file v4.3 and there are no supported formats for it. These are the available formats:
`[com.sun.tdk.signaturetest.sigfile.F41Format@40cb8df7, com.sun.tdk.signaturetest.sigfile.F21Format@13b13b5d, com.sun.tdk.signaturetest.sigfile.F31Format@2892dae4, com.sun.tdk.signaturetest.sigfile.F40Format@355ce81c]`

That 1.4 seems to be the latest version: https://repo1.maven.org/maven2/org/netbeans/tools/sigtest-maven-plugin/

With JDK8 I get an strange error:
`[ERROR] Failed to execute goal on project jakarta.mail-tck: Could not resolve dependencies for project com.sun.mail:jakarta.mail-tck:jar:1.0.0-SNAPSHOT: Could not find artifact org.netbeans.tools:ct-sym:jar:latest at specified path /home/jbescos/programs/java/jdk1.8.0_231/jre/lib/ct.sym`

Reply:
> That version 1.4 does not work with the signature file of JDK11. I get this:
> 
> ```
> [ERROR] sigtest.SignatureTest.test  Time elapsed: 0.057 s  <<< FAILURE!
> org.opentest4j.AssertionFailedError: 
> /jakarta.mail_2.0_java_11.sig
> Can not find or read signature file(s): /tmp/jakarta.mail_2.0_java_11.sig
> ```
> 
> I have debugged it, and the problem is that the signature file is` #Signature file v4.3` and there are no supported formats for it. These are the available formats:
> `[com.sun.tdk.signaturetest.sigfile.F41Format@40cb8df7, com.sun.tdk.signaturetest.sigfile.F21Format@13b13b5d, com.sun.tdk.signaturetest.sigfile.F31Format@2892dae4, com.sun.tdk.signaturetest.sigfile.F40Format@355ce81c]`
> 
> That 1.4 seems to be the latest version: https://repo1.maven.org/maven2/org/netbeans/tools/sigtest-maven-plugin/
> 
> With JDK8 I get an strange error:
> `[ERROR] Failed to execute goal on project jakarta.mail-tck: Could not resolve dependencies for project com.sun.mail:jakarta.mail-tck:jar:1.0.0-SNAPSHOT: Could not find artifact org.netbeans.tools:ct-sym:jar:latest at specified path /home/jbescos/programs/java/jdk1.8.0_231/jre/lib/ct.sym`

https://github.com/eclipse-ee4j/jakartaee-tck/pull/720 contains the Platform TCK changes we made to use the 1.4 artifact.

1. https://github.com/eclipse-ee4j/jakartaee-tck/pull/720/files#diff-9dd8b510811e56b95d7756a4997a46d72403b143830646a1d9e1bf544e39fbabR48 shows how we switched to only use the Java SE 8 signature map files which should help address the failure finding JDK11 signature map file.
2. https://github.com/eclipse-ee4j/jakartaee-tck/pull/720/files#diff-5e4cca3633f2f77f8643df53f50a4ba84f69fcf6073a1cec857b00f63b1f9afcR1 shows the Jakarta Mail signature map file (I generated it with the 1.4 sigtest artifact but you could try copying https://raw.githubusercontent.com/scottmarlow/jakartaee-tck/88bef553993a232f0800f20c6d839c89c568e345/src/com/sun/ts/tests/signaturetest/signature-repository/jakarta.mail.sig_2.0_se8 into the Jakarta Mail TCK and see if that works).
3. I didn't see the strange error locally with JDK8 testing.  
4. I think you need add `-IgnoreJDKClass` option to Mail TCK root build.xml file, see https://github.com/eclipse-ee4j/jakartaee-tck/pull/720/files#diff-1a4744864acc9458a363b918e0971b0a7b9710f878cb23078d7feff634227333R65 
5. Note that jakarta.resource.spi.SecurityException should be removed from `IgnoreJDKClass` options passed in `build.xml` as was done via https://github.com/eclipse-ee4j/jakartaee-tck/commit/439419b90eddf8d04b75adfb8b0001f093285920

Reply:
Now for JDK 8 it is good: https://github.com/eclipse-ee4j/mail-tck/pull/29/commits/2af4fbf543c422b2a971c97f535cfd59f5c57e84

I had also to exclude `org.netbeans.tools:ct-sym` because it is a system scope dependency: https://repo1.maven.org/maven2/org/netbeans/tools/sigtest-maven-plugin/1.4/sigtest-maven-plugin-1.4.pom

Reply:
I expect that you will see failures on JDK11+ due to not yet adding the -IgnoreJDKClass option as mentioned above.

Reply:
In JDK 11+ it is still failing because of the format error related to `#Signature file v4.3`. I have only updated the sig file of JDK 8 from here https://raw.githubusercontent.com/scottmarlow/jakartaee-tck/88bef553993a232f0800f20c6d839c89c568e345/src/com/sun/ts/tests/signaturetest/signature-repository/jakarta.mail.sig_2.0_se8 

The `-IgnoreJDKClass` option is added in the SignatureTest.java. I don't have a build.xml.

If I change it to use the jakarta.mail.sig_2.0_se8  when running JDK11, I am getting the same error than in JDK8:

```
[ERROR] sigtest.SignatureTest.test  Time elapsed: 0.484 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: 
/jakarta.mail_2.0_java_1.8.sig
SignatureTest report
Base version: 1.0
Tested version: 
Check mode: src [throws normalized]
Constant checking: on


Added Fields
------------

jakarta.mail.Session:                   field public final static jakarta.mail.util.StreamProvider jakarta.mail.Session.STREAM_PROVIDER

Added Methods
-------------

jakarta.mail.internet.MimeUtility:      method public static byte[] jakarta.mail.internet.MimeUtility.getBytes(java.io.InputStream) throws java.io.IOException
jakarta.mail.internet.MimeUtility:      method public static byte[] jakarta.mail.internet.MimeUtility.getBytes(java.lang.String)


 ==> expected: <true> but was: <false>
	at sigtest.SignatureTest.test(SignatureTest.java:76)

```

Reply:
I think this failure is expected - these were added in 2.1 (sig file will have to be updated to accommodate this at some point)

Reply:
hmm but do we need that field? should it be `private` instead? It is not something we want to expose through the API

Reply:
> hmm but do we need that field? should it be `private` instead? It is not something we want to expose through the API

Here it is the modification: https://github.com/eclipse-ee4j/mail/pull/579